include/deal.II/base/polynomials_hermite.h

```diff
@@ -0,0 +1,152 @@
+// ---------------------------------------------------------------------
+//
+// Copyright (C) 2021 - 2022 by the deal.II authors
+//
+// This file is part of the deal.II library.
+//
+// The deal.II library is free software; you can use it, redistribute
+// it, and/or modify it under the terms of the GNU Lesser General
+// Public License as published by the Free Software Foundation; either
+// version 2.1 of the License, or (at your option) any later version.
+// The full text of the license can be found in the file LICENSE.md at
+// the top level directory of deal.II.
+//
+// ---------------------------------------------------------------------
+
+
+
+#ifndef dealii_hermite_polynomials
+#define dealii_hermite_polynomials
+
+#include <deal.II/base/config.h>
+
+#include <deal.II/base/exceptions.h>
+#include <deal.II/base/point.h>
+#include <deal.II/base/polynomial.h>
+#include <deal.II/base/subscriptor.h>
+
+#include <vector>
+
+
+
+DEAL_II_NAMESPACE_OPEN
+
+/**
+ * @addtogroup Polynomials
+ * @{
+ */
+
+namespace Polynomials
+{
+  /**
+   * This class implements Hermite interpolation polynomials (see
+   * @cite CiarletRiavart1972interpolation) enforcing the maximum
+   * possible level of regularity $r$ in the FEM basis given a
+   * polynomial degree of $2r+1$. The polynomials all represent
+   * either a non-zero shape value or derivative at $x=0$ and $x=1$
+   * on the reference interval $x \in [0,1]$.
+   *
+   * Indices $j = 0, 1, \dots, r$ refer to polynomials corresponding
+   * to a non-zero derivative (or shape value for $j=0$) of
+   * order $j$ at $x=0$, and indices $j = r+1, r+2, \dots, 2r+1$
+   * refer to polynomials with a non-zero derivative of order
+   * $j-(r+1)$ (or value for $j=r+1$) at $x=1$. In particular, the
+   * $0^{th}$ function has a value of $1$ at $x=0$, and the
+   * $(r+1)^{th}$ function has a value of $1$ at $x=1$.The basis is
+   * rescaled such that a function corresponding to a non-zero $j^{th}$
+   * derivative has derivative value $j! 4^{j}$ at the corresponding
+   * node. This is done to prevent the $L^{2}$-norm of the basis functions
+   * from reducing exponentially with the chosen regularity.
+   */
+  class PolynomialsHermite : public Polynomial<double>
```

Comment:
While `PolynomialsHermite` is the right name if we were starting from scratch, we already have `HermiteInterpolation` and `HermiteLikeInterpolation` in the library. Could you pick a name which better reflects the fact that this one (unlike the others) actually implements higher regularity?

Reply:
Let me amend this comment - Peter mentioned to me that someone else suggested this name so lets go with it (ultimately it's just a name and not that important).